llvm/lib/Transforms/AggressiveInstCombine/AggressiveInstCombine.cpp

```diff
@@ -1457,6 +1710,9 @@ static bool foldUnusualPatterns(Function &F, DominatorTree &DT,
       MadeChange |= foldConsecutiveLoads(I, DL, TTI, AA, DT);
       MadeChange |= foldPatternedLoads(I, DL);
       MadeChange |= foldICmpOrChain(I, DL, TTI, AA, DT);
+      MadeChange |= foldMul128Low(I, DL, DT);
```

Comment:
Maybe gate on i128 legality? Otherwise, It may lead to regressions on targets without native i128 mul.